### PR TITLE
MAINT: Keep applying ruff/flake8-bugbear rules (B)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Follow-up to #26902.

Apply additional rules in preparation for https://github.com/numpy/numpy/issues/24994.

### CODE REVIEW

File: numpy/_core/tests/test_simd.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def test_operators_logical(self):
         assert vor == data_or
 
         data_xor = [a ^ b for a, b in zip(data_a, data_b)]
-        vxor = getattr(self, "xor")(vdata_a, vdata_b)
+        vxor = self.xor(vdata_a, vdata_b)
         assert vxor == data_xor
 
         vnot = getattr(self, "not")(vdata_a)
```
Comment (charris) [inline numpy/_core/tests/test_simd.py:166]: I wonder why this was not changed.

**Review by charris — APPROVED** [review]

File: numpy/_core/tests/test_simd.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply numpy/_core/tests/test_simd.py:166]: I guess this is an unusual construct that is currently not caught by ruff.

Comment (mhvk) [reply numpy/_core/tests/test_simd.py:166]: I think it is because one cannot do `class A: not = bla` - it is a reserved keyword (unlike `xor`).

**Review by mhvk — APPROVED** [review]
Nice changes again!

File: doc/neps/tools/build_index.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def nep_metadata():
 
             # The title should be the first line after a line containing only
             # * or = signs.
-            for i, line in enumerate(lines[:-1]):
+            for line in lines[:-1]:
```
Comment (charris) [inline doc/neps/tools/build_index.py:41]: Hmm,  ` i` is used in line 48. I don't know if this should be classed as a ruff bug, or obscure programming. Apparently the `i` is not local to the `for` statement.

Comment (charris) [reply doc/neps/tools/build_index.py:41]: I think this is a ruff bug.

Comment (DimitriPapadopoulos) [reply doc/neps/tools/build_index.py:41]: Indeed, `enumerate` shouldn't have been removed here, I'll revert.

This is not necessarily a ruff bug, in the sense that I had run it with `--fix --unsafe-fixes`. I hadn't noticed the later use of `i` when reviewing the changes.

**Pull request merged**